openslides_backend/action/action_worker.py

```diff
@@ -0,0 +1,288 @@
+        with self.datastore.get_database_context():
+            if not self.new_id:
+                self.new_id = self.datastore.reserve_id(collection="action_worker")
+                self.fqid = fqid_from_collection_and_id("action_worker", self.new_id)
+            try:
+                self.datastore.write_action_worker(
+                    WriteRequest(
+                        events=[
+                            Event(
+                                type=EventType.Create,
+                                fqid=self.fqid,
+                                fields={
+                                    "id": self.new_id,
+                                    "name": self.action_names,
+                                    "state": "running",
+                                    "created": self.starttime,
+                                    "timestamp": current_time,
+                                },
+                            )
+                        ],
+                        information={self.fqid: ["create action_worker"]},
+                        user_id=self.user_id,
+                        locked_fields={},
+                    )
+                )
+                self.datastore.get(
+                    self.fqid, [], lock_result=False, use_changed_models=False
+                )
+                message = f"Action lasts too long. {self.fqid} written to database. Get the result from database, when the job is done."
+                self.written = True
+            except DatastoreException as e:
+                message = f"Action lasts too long, exception on writing {self.fqid}: {e.message}. Get the result later from database."
+        self.logger.debug(f"action_worker: {message}, written:{self.written}")
+        return message
+
+    def continue_action_worker_write(self) -> None:
+        current_time = round(time())
+        with self.datastore.get_database_context():
+            ram = psutil.virtual_memory()
+            response = {
+                "success": False,
+                "message": f"ram total:{ram.total} available:{ram.available} percent:{ram.percent} used:{ram.used} free.{ram.free}",
+                "action_error_index": 0,
+                "action_data_error_index": 0,
+            }
+            self.datastore.write_action_worker(
+                WriteRequest(
+                    events=[
+                        Event(
+                            type=EventType.Update,
+                            fqid=self.fqid,
+                            fields={
+                                "timestamp": current_time,
+                                "result": response,
+                            },
+                        )
+                    ],
+                    information={self.fqid: ["timestamp during running action_worker"]},
+                    user_id=self.user_id,
+                    locked_fields={},
+                )
+            )
+            self.logger.debug(
```

Comment:
Mmh, don't you think that this is too much spam? Printing the current time is also not really meaningful. But the message itself might help to determine whether the thread is still running or not, I'm not sure.

Reply:
All the action_worker writing will be made only by very rare actions, the big majority of them don't need a second and will not use action_worker-notifiers. This intermediate update-writing appears after 11sec for the first time. That are not common actions using the action_worker-notifyings.

Reply:
ok